site/src/pages/ManagementSettingsPage/IdpSyncPage/IdpSyncPageView.tsx

```diff
@@ -46,76 +85,118 @@ export const IdpSyncPageView: FC<IdpSyncPageViewProps> = ({ oidcConfig }) => {
 					/>
 				</Cond>
 				<Cond>
-					<Stack spacing={2} css={styles.fields}>
-						{/* Semantically fieldset is used for forms. In the future this screen will allow
-						 updates to these fields in a form */}
-						<fieldset css={styles.box}>
-							<legend css={styles.legend}>Groups</legend>
-							<Stack direction={"row"} alignItems={"center"} spacing={8}>
-								<IdpField
-									name={"Sync Field"}
-									fieldText={groups_field}
-									showStatusIndicator
-								/>
-								<IdpField
-									name={"Regex Filter"}
-									fieldText={group_regex_filter}
-								/>
-								<IdpField
-									name={"Auto Create"}
-									fieldText={group_auto_create?.toString()}
-								/>
-							</Stack>
-						</fieldset>
-						<fieldset css={styles.box}>
-							<legend css={styles.legend}>Roles</legend>
-							<Stack direction={"row"} alignItems={"center"} spacing={3}>
-								<IdpField
-									name={"Sync Field"}
-									fieldText={user_role_field}
-									showStatusIndicator
-								/>
-							</Stack>
-						</fieldset>
-					</Stack>
-					<Stack spacing={6}>
-						<IdpMappingTable
-							type="Role"
-							isEmpty={Boolean(
-								!oidcConfig?.user_role_mapping ||
-									Object.entries(oidcConfig?.user_role_mapping).length === 0,
-							)}
+					<>
+						<Tabs
+							active={tab}
+							css={{
+								marginBottom: 20,
+							}}
 						>
-							{oidcConfig?.user_role_mapping &&
-								Object.entries(oidcConfig.user_role_mapping)
-									.sort()
-									.map(([idpRole, roles]) => (
-										<RoleRow
-											key={idpRole}
-											idpRole={idpRole}
-											coderRoles={roles}
+							<TabsList>
+								<TabLink to="?tab=groups" value="groups">
+									Group Sync Settings
+								</TabLink>
+								<TabLink to="?tab=roles" value="roles">
+									Role Sync Settings
+								</TabLink>
+							</TabsList>
+						</Tabs>
+						{tab === "groups" ? (
+							<>
+								<div css={styles.fields}>
+									<Stack direction={"row"} alignItems={"center"} spacing={6}>
+										<IdpField
+											name={"Sync Field"}
+											fieldText={groupSyncSettings?.field}
+											showStatusIndicator
 										/>
-									))}
-						</IdpMappingTable>
-						<IdpMappingTable
-							type="Group"
-							isEmpty={Boolean(
-								!oidcConfig?.group_mapping ||
-									Object.entries(oidcConfig?.group_mapping).length === 0,
-							)}
-						>
-							{oidcConfig?.user_role_mapping &&
-								Object.entries(oidcConfig.group_mapping)
-									.sort()
-									.map(([idpGroup, group]) => (
-										<GroupRow
-											key={idpGroup}
-											idpGroup={idpGroup}
-											coderGroup={group}
+										<IdpField
+											name={"Regex Filter"}
+											fieldText={
+												typeof groupSyncSettings?.regex_filter === "string"
+													? groupSyncSettings?.regex_filter
+													: "none"
```

Comment:
The second `?.` should be safe to remove, since the `typeof` check will have already proven to the compiler that the value is both a string and defined